src/plugins/discover/public/application/apps/main/utils/update_search_source.ts

```diff
@@ -6,48 +6,47 @@
  * Side Public License, v 1.
  */
 
-import { getSortForSearchSource } from '../angular/doc_table';
-import { SAMPLE_SIZE_SETTING, SORT_DEFAULT_ORDER_SETTING } from '../../../common';
-import { IndexPattern, ISearchSource } from '../../../../data/common/';
-import { SortOrder } from '../../saved_searches/types';
-import { DiscoverServices } from '../../build_services';
+import { getSortForSearchSource } from '../../../angular/doc_table';
+import { SAMPLE_SIZE_SETTING, SORT_DEFAULT_ORDER_SETTING } from '../../../../../common';
+import { IndexPattern, ISearchSource, SearchSource } from '../../../../../../data/common';
+import { SortOrder } from '../../../../saved_searches/types';
+import { DiscoverServices } from '../../../../build_services';
+import { indexPatterns as indexPatternsUtils } from '../../../../../../data/public';
 
 /**
  * Helper function to update the given searchSource before fetching/sharing/persisting
  */
-export function updateSearchSource({
-  indexPattern,
-  services,
-  sort,
-  columns,
-  useNewFieldsApi,
-  showUnmappedFields,
-  persistentSearchSource,
-  volatileSearchSource,
-}: {
-  indexPattern: IndexPattern;
-  services: DiscoverServices;
-  sort: SortOrder[];
-  columns: string[];
-  useNewFieldsApi: boolean;
-  showUnmappedFields?: boolean;
-  persistentSearchSource: ISearchSource;
-  volatileSearchSource?: ISearchSource;
-}) {
+export function updateSearchSource(
+  searchSource: ISearchSource,
+  persist = true,
+  {
+    indexPattern,
+    services,
+    sort,
+    useNewFieldsApi,
+  }: {
+    indexPattern: IndexPattern;
+    services: DiscoverServices;
+    sort: SortOrder[];
+    useNewFieldsApi: boolean;
+  }
+) {
   const { uiSettings, data } = services;
   const usedSort = getSortForSearchSource(
     sort,
     indexPattern,
     uiSettings.get(SORT_DEFAULT_ORDER_SETTING)
   );
+  const usedSearchSource = persist ? searchSource : (searchSource!.getParent() as SearchSource);
```

Comment:
Removing the unecessary non-null assertion and expressing a bit better our intention (since we're just wanting to assume non-null here, knowing that the volatile search source will always have a parent):

```suggestion
  const usedSearchSource = persist ? searchSource : searchSource.getParent()!;
```

Maybe also worth adding this as a note why we do the non-null assertion here, like:

```
We know the volatile search source we pass in when `persist === false` will always have a parent.
```

Reply:
nice & true, thx